Enable alpha channel in color selector

adanielyan opened the pull request:
This pull request adds alpha channel to Color interface in a form of 4th text box.

![image](https://user-images.githubusercontent.com/861777/139356335-eb89da7e-40fb-44a7-b609-39dc2776faec.png)

paulboudewijn committed:
Fix localstorage file deletion (#9126)

* excluded VS specific files

* new files after updating directus sources

* case insensitive string matching fix

* Remove accidental commit files

* Cleanup gitignore

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update dependency vite to v2.6.11 (#9121)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency axios to v0.24.0 (#9132)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

rijkvanzanten committed:
New Crowdin updates (#9138)

* Update source file en-US.yaml

* New translations en-US.yaml (Serbian (Latin))

* New translations en-US.yaml (Portuguese, Brazilian)

* New translations en-US.yaml (Indonesian)

* New translations en-US.yaml (Spanish, Chile)

* New translations en-US.yaml (Thai)

* New translations en-US.yaml (Hindi)

* New translations en-US.yaml (Chinese Traditional)

* New translations en-US.yaml (Spanish, Latin America)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Portuguese)

* New translations en-US.yaml (Swedish)

* New translations en-US.yaml (Turkish)

* New translations en-US.yaml (Estonian)

* New translations en-US.yaml (Vietnamese)

* New translations en-US.yaml (Chinese Simplified)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Finnish)

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Arabic)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Catalan)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Hungarian)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Japanese)

* New translations en-US.yaml (Lithuanian)

* New translations en-US.yaml (Dutch)

* New translations en-US.yaml (Norwegian)

* New translations en-US.yaml (Slovenian)

renovate[bot] committed:
Update dependency @types/busboy to v0.3.1 (#9096)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency @types/js-yaml to v4.0.4 (#9090)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency sass to v1.43.3 (#9035)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

benhaynes committed:
Docs structure (#9071)

* WIP

* updates

* docs updates

* structure

* big structure update

* docs module icon change

* in-app docs nav

* more content and structure changes

* Remove redundant

* Fix docs build in app

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update dependency stylelint-config-standard to v23 (#9026)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency stylelint to v14 (#9016)

* Update dependency stylelint to v14

* Update dependency stylelint-scss to v4

* Update dependency stylelint-order to v5

* Undo command change

* Update stylelint command

* Use modern color syntax

Who knew this was already supported everywhere? Awesome!!

* Update stylelint-config-standard

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Pin dependency stylelint-config-standard to 23.0.0 (#9140)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

rijkvanzanten committed:
Fix type issues in busboy update

oreilles committed:
Change v-checkbox background color when disabled (#9070)

* Change v-checkbox background color when disabled

* Only apply style for block input.

rijkvanzanten committed:
Don't enforce rgb modern syntax

azrikahar committed:
Use textarea input for textarea placeholder (#9041)

AustinPhillipTaylor committed:
Reworked Sass to automatically generate color variations from palette (#8890)

* Reworked Sass to automatically generate color variations from palette

* Fixed naming scheme for mixing colors

* Updated light theme to use pre-defined steps

* Updated dark theme to use pre-defined steps, added steps at 90 & 110

* Hardcoded -alt color variations, extracted color variation generator to reusable mixin.

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update dependency ts-node to v10.4.0 (#9083)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency eslint to v8.1.0 (#9091)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency oracledb to v5.3.0 (#9042)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update typescript-eslint monorepo to v5.2.0 (#9133)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

azrikahar committed:
fix time series date field allow list (#9143)

paescuj committed:
Wait for checks before auto-merge Crowdin updates (#9156)

renovate[bot] committed:
Update dependency lint-staged to v11.2.5 (#9155)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

aidenfoxx committed:
Ignore email field in oauth and store email in external_identifier (#9153)

* Ignore email field in oauth and store email in external_identifier if needed

* Removed unused variable

joselcvarela committed:
docs: improve running locally (#9142)

renovate[bot] committed:
Update gatsby monorepo to v4 (#9028)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Jay Cammarano <jay.cammarano@gmail.com>

renovate[bot] committed:
Update dependency micromark to v3.0.7 (#9141)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

rijkvanzanten committed:
Add missing config options file

rijkvanzanten committed:
Fix link/method ref

rijkvanzanten committed:
Remove homepage redirect

benhaynes committed:
Add padding to calendar layout event

renovate[bot] committed:
Update dependency stylelint to v14.0.1 (#9168)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency vite to v2.6.12 (#9169)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

rijkvanzanten committed:
Re-add missing oauth docs (#9170)

renovate[bot] committed:
Update Node.js to v16.13.0 (#9174)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

aidenfoxx committed:
Added the provider to auth hooks (#9059)

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update dependency knex-schema-inspector to v1.6.4 (#9159)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency sass to v1.43.4 (#9177)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency lint-staged to v11.2.6 (#9175)

nickrum committed:
Set no-cache header on extension sources (#9186)

This should hopefully fix caching issues some users were experiencing.

nickrum committed:
Clean up App type shims (#9190)

nickrum committed:
Clean up App base url replacement (#9192)

grefrit committed:
removed toLowerCase for dbSafe fields (#9183)

oreilles committed:
Add `folder`, `template`, `enableCreate`, `enableSelect` options to the files interface (#9184)

oreilles committed:
Always show folder icon for alias tables (folders) (#9187)

* Always show folder icon for alias tables (folders)

* Update app/src/modules/settings/routes/data-model/collections/components/collection-item.vue

Co-authored-by: Nicola Krumschmidt <nicola.krumschmidt@freenet.de>

Co-authored-by: Nicola Krumschmidt <nicola.krumschmidt@freenet.de>

oreilles committed:
Fixed dynamic translations import. (#9179)

jamescammarano committed:
Fixes broken links in docs (#9172)

* first half of broken links fixed

* next chunk of broken links resolved

* next chunk of urls

* last chunk of links

* one more

* filter rules

* homepage links

* removed localhost

* filter rules for query

* updated links

* relationships link

* changed links from references to glossary

nickrum committed:
Export API dist folder at the root of the package path (#9199)

aidenfoxx committed:
Clarified some oauth things in the docs (#9203)

renovate[bot] committed:
Update dependency vite to v2.6.13 (#9193)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency pinia to v2.0.0 (#9205)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency simple-git-hooks to v2.7.0 (#9195)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency @vitejs/plugin-vue to v1.9.4 (#9181)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

nickrum committed:
Force renovate to use Node 16.13 (#9209)

nickrum committed:
Revert "Force renovate to use Node 16.13 (#9209)" (#9219)

benhaynes committed:
Update popular links

Commit:
removed unused code

RGBA-> RGB(A)

added alpha slider

added label to alpha

renovate[bot] committed:
Update dependency vite-plugin-md to v0.11.4 (#9206)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

azrikahar committed:
tweak fallback interface selectors (#9234)

benhaynes committed:
update field flow styling

Jakob-em committed:
Add configurable headers for webhooks (#8855)

* Add configurable headers for webhooks

* Update api/src/database/migrations/20211016A-add-webhook-headers.ts

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

benhaynes committed:
broken database docs mirroring link

aidenfoxx committed:
LDAP auth provider (#9074)

* Added LDAP implementation

* Cleanup and added refresh checking

* Added provider check to ldap

* Added LDAP docs

* Always update LDAP user role on authentication

* Fetched package-json from github

* Fixed lint error

* Update api/src/auth/drivers/ldap.ts

* Update api/src/auth/drivers/ldap.ts

* Added missing locale key

* Update docs/guides/api-config.md

* Update docs/configuration/config-options.md

* Update api/src/auth/drivers/ldap.ts

* Added back LDAP example

* Update config-options.md

* getUserID function sjhouldn't care about password

* Added LOCKOUT to INVALID_ACCOUNT_FLAGS

* Update styling approach for v-select

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

danilopolani committed:
Append `access_token` to images in WYSIWYG only once (#9062)

* append access_token to image in wysiwyg only once

* do not append token in previewUrl when clicking an image from wysiwyg

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Pin dependency @types/ldapjs to 2.2.1 (#9238)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

oreilles committed:
Fix generated columns being required.  (#9200)

* Fix generated columns being required. Also prevent schema changes on generated columns.

* Fix type errors

* Disable `unique` and `nullable` instead of not showing them.

* Fix other type error

* Nullable check small refactor

* Fixed MSSQL query

* Fixed oracle query

oreilles committed:
Map interface improvements (#9239)

* Add tooltips on feature hover on the interface
* Add marker on geocoder search result
* Improve map interface style

oreilles committed:
Fix dynamic variables not working anymore in UUID filter (#9180)

* Fix $CURRENT_USER not working anymore in UUID filter

* $Handle CURRENT_ROLE too

benhaynes committed:
Fixes for docs links

jaycammarano commented:
should the bar be labeled? I wouldn't know what that was right now.

adanielyan commented:
> should the bar be labeled? I wouldn't know what that was right now.

It could be, not sure where it will be the best to put the label though. On the other hand, it should not take a lot of effort to figure out what this slider does, even without a label.

adanielyan commented:
Does this look better?

![image](https://user-images.githubusercontent.com/861777/139483387-9f33ce88-72ad-4837-8e0d-d55040389dd9.png)

rijkvanzanten committed:
Add CLI to exports (#9241)

jaycammarano commented:
> Does this look better?
> 
> ![image](https://user-images.githubusercontent.com/861777/139483387-9f33ce88-72ad-4837-8e0d-d55040389dd9.png)

I like that.

rijkvanzanten commented:
> My understanding is that alpha is commonly set as 0 to 100 and stored as rgba(r, g, b, a), though I realize there is also this hex spec.

As a matter of fact, that spec recently got updated, and is now `rgb(r g b a)` where a is 0-100 instead of 0-1 IIRC.

Regardless, opacity should be 0-100, not 0-255 🙂

Commit:
merge

adanielyan commented:
> > My understanding is that alpha is commonly set as 0 to 100 and stored as rgba(r, g, b, a), though I realize there is also this hex spec.
> 
> As a matter of fact, that spec recently got updated, and is now `rgb(r g b a)` where a is 0-100 instead of 0-1 IIRC.
> 
> Regardless, opacity should be 0-100, not 0-255 🙂

I think 1-100 is what I am used to as well. The only reason for changing it to 255 was that when converted to hex, 1-100 cannot cover the entire range of 00-FF values while 0-255 can. So 1-100 range will cover only 2/5 of all possible values. That said I can easily change it back to 0-100 if you'd like.

rijkvanzanten committed:
v9.0.0-rc.100

rijkvanzanten committed:
Update changelog.md

Commit:
change alpha back to 0-100

rijkvanzanten committed:
Update package.json

rijkvanzanten committed:
Update package.json

aidenfoxx committed:
Fixed Github oauth config (#9256)

oreilles committed:
Removed invalid column in mssql schema inspector (#9260)

jamescammarano committed:
workflow-dispatch added to e2e-tests.yml (#9265)

danilopolani committed:
fix new field group selection (#9301)

dimitrov-adrian committed:
Add shadows to v-menu angles (#9297)

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

paescuj committed:
Revise GitHub workflows #2 (#9240)

* GitHub workflow revision #2

* Add a few comments

* Add and fix comments

* Move kodiak config into .github

To not overflow the root directory

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

paescuj committed:
Rearrange on events in e2e workflow (#9308)

rijkvanzanten committed:
New Crowdin updates (#9158)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (French)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Estonian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Vietnamese)

* New translations en-US.yaml (Vietnamese)

* New translations en-US.yaml (Breton)

* New translations en-US.yaml (Norwegian)

* New translations en-US.yaml (Norwegian)

* New translations en-US.yaml (Serbian (Latin))

* New translations en-US.yaml (Portuguese, Brazilian)

* New translations en-US.yaml (Indonesian)

* New translations en-US.yaml (Spanish, Chile)

* New translations en-US.yaml (Thai)

* New translations en-US.yaml (Malay)

* New translations en-US.yaml (Breton)

* New translations en-US.yaml (Chinese Traditional)

* New translations en-US.yaml (Spanish, Latin America)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Swedish)

* New translations en-US.yaml (Turkish)

* New translations en-US.yaml (Estonian)

* New translations en-US.yaml (Vietnamese)

* New translations en-US.yaml (Chinese Simplified)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Finnish)

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Arabic)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Catalan)

* New translations en-US.yaml (Czech)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Greek)

* New translations en-US.yaml (Ukrainian)

* New translations en-US.yaml (Hungarian)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Japanese)

* New translations en-US.yaml (Lithuanian)

* New translations en-US.yaml (Dutch)

* New translations en-US.yaml (Norwegian)

* New translations en-US.yaml (Slovenian)

* Update source file en-US.yaml

* New translations en-US.yaml (Romanian)

* New translations en-US.yaml (Serbian (Latin))

* New translations en-US.yaml (Portuguese, Brazilian)

* New translations en-US.yaml (Indonesian)

* New translations en-US.yaml (Persian)

* New translations en-US.yaml (Spanish, Chile)

* New translations en-US.yaml (Thai)

* New translations en-US.yaml (Malay)

* New translations en-US.yaml (Bengali, India)

* New translations en-US.yaml (Chinese Traditional)

* New translations en-US.yaml (Spanish, Latin America)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Slovak)

* New translations en-US.yaml (Serbian (Cyrillic))

* New translations en-US.yaml (Swedish)

* New translations en-US.yaml (Turkish)

* New translations en-US.yaml (Estonian)

* New translations en-US.yaml (Vietnamese)

* New translations en-US.yaml (Chinese Simplified)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Finnish)

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Afrikaans)

* New translations en-US.yaml (Arabic)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Catalan)

* New translations en-US.yaml (Czech)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Greek)

* New translations en-US.yaml (Hebrew)

* New translations en-US.yaml (Ukrainian)

* New translations en-US.yaml (Hungarian)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Japanese)

* New translations en-US.yaml (Georgian)

* New translations en-US.yaml (Lithuanian)

* New translations en-US.yaml (Dutch)

* New translations en-US.yaml (Norwegian)

* New translations en-US.yaml (Slovenian)

* Update source file en-US.yaml

* Update source file en-US.yaml

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Chinese Traditional)

* New translations en-US.yaml (Chinese Traditional)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

aidenfoxx committed:
Removed undefined type from LDAP userAccountControl (#9309)

dorianim committed:
Feat/custom ldap mail attribute (#9307)

* Allow custom email field for LDAP

* Update docs

* Break out into variable and don't cast to String

oreilles committed:
Set calendar height to 100% (#9295)

* Set calendar height to 100% and set `smallHeader to true

* Unset `smallHeader`

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

aidenfoxx committed:
Update Oracle to support `is_generated` (#9310)

paescuj committed:
Switch back to fkirc/skip-duplicate-actions (#9312)

All changes have been merged in

renovate[bot] committed:
Update dependency npm to v8.1.2 (#9221)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

rijkvanzanten committed:
New translations en-US.yaml (German) (#9316)

renovate[bot] committed:
Update dependency knex to v0.95.12 (#9208)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

aidenfoxx committed:
Added state param to oauth to make Okta happy (#9289)

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update typescript-eslint monorepo to v5.3.0 (#9317)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

danilopolani committed:
prefix aggregation field with table name (#9314)

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update dependency rollup to v2.59.0 (#9293)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

jamescammarano committed:
End-to-end tests for filters (#9078)

* reorganized for sanity

* filter tests for _eq/_neq

* logical operators

* update one one to many

* /:collection PATCH one to many

* base tables.id => uuids to minimize collisions

* tests pass

* added python to dockerfile

* tests passing

* ci?

* ci...

* hanging async

azrikahar committed:
Input code json (#9291)

* remove language selection for input code type JSON

* lint as json when type is json

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

jamescammarano committed:
Add SET NULL to directus_files.uploaded_by constraint (#9305)

* add set null to directus_files constraint

* SET NULL on created_by and modified_by

* "user_created" and "user_updated" => SET NULL

* SET NULL on collections optional system fields

* fixed dropColumn()

* Add line breaks, cause little OCD

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

azrikahar committed:
Update translation icon in interface selector (#9292)

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

oreilles committed:
Map layout and interface improvements (#9288)

* Map layout and interface improvements
* Fix marker not showing up on geocoder search
* Replaced geocoder search placeholder
* Fix geocoder hit area
* Fix item popup positioning
* Removed unselect button
* Removed "No results" popup
* Removed option to filter map on demand vs automatically
* Renamed Geometry field option
* Added placeholder to template option
* Hide "Delete" button when no feature are selected

* Lint fix

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

benhaynes committed:
add content and fix broken links (#9321)

Nitwel committed:
Fix issues with fields config (#9324)

* Fix issues with fields config

* Fix /fields endpoint return type for alias fields

* Fix display selection in relational interfaces

jamescammarano committed:
fixes many dead links (#9325)

rijkvanzanten committed:
New Crowdin updates (#9326)

* Update source file en-US.yaml

* New translations en-US.yaml (Portuguese, Brazilian)

* New translations en-US.yaml (Spanish, Chile)

* New translations en-US.yaml (Thai)

* New translations en-US.yaml (Spanish, Latin America)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Chinese Simplified)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Arabic)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Catalan)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Hungarian)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Slovenian)

oreilles committed:
Added supported function to geometryHelper, added geometrySupport to serverinfo (#9290)

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

azrikahar committed:
fix lint warnings (#9287)

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

jamescammarano committed:
Fixes broken docs links (#9327)

* a few more dead links

* one last link set

* fixed bad link

azrikahar committed:
filter files by folder when a folder is specified (#9285)

azrikahar committed:
Prevent file input preloading drawer (#9282)

azrikahar committed:
improve disabled interface selector style (#9281)

joselcvarela committed:
Fix: disable sort field for singleton collections (#9270)

* disable sort for singleton collections

* fix modify sort then activate singleton

oreilles committed:
Fixed nullable and unique disabled check (#9269)

azrikahar committed:
Context menu improvements (#9271)

* add focusout to close context menus

* add "Edit collection" to collections context menu

* disable context menu when nothing to show

* remove obsolete code

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

8byr0 committed:
fix(graphql): remove __typename from selection nodes when present (#9318)

* fix(graphql): remove __typename from selection nodes when present

* fix: cast selectionNode to FieldNode in filter callback

rijkvanzanten committed:
New translations en-US.yaml (Italian) (#9330)

rijkvanzanten committed:
New translations en-US.yaml (Arabic) (#9332)

rijkvanzanten committed:
New translations en-US.yaml (Breton) (#9335)

azrikahar committed:
fix m2a not savable (#9349)

d1rOn committed:
Translations interface fix (#9333)

* Translations interface fix

Fixes issue

* Translations interface fix (2)

paescuj committed:
Fix eslint warnings (#9357)

jamescammarano committed:
Fixes broken links in the documentation (#9354)

* broken links

* more broken links

* last but not least

renovate[bot] committed:
Update dependency @types/sharp to v0.29.3 (#9352)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

paulboudewijn committed:
Fix nullable boolean (#8497)

* nullable boolean can now be set to null again after it weas true or false

* Implementation changed to Composition API

* Implementation changed to Composition API

* fixed formatting issues

* fixed formatting issues

* fixed formatting issues

* fixed formatting issues

* run linters on branch

* run linters on branch

* run linters on branch

* run linters on branch

* fixed formatting issues

* changes to checkbox interface reverted

* changes to checkbox interface reverted

* Update boolean.vue

Nitwel committed:
Fix date on sqlite (#7774)

* fix date on sqlite

* remove unused imports

* rename KnexSpatial to KnexDate

Co-authored-by: Jose Varela <joselcvarela@gmail.com>

rijkvanzanten committed:
New translations en-US.yaml (Polish) (#9361)

aidenfoxx committed:
Added openid fallback to user token info if profile URL not defined (#9368)

renovate[bot] committed:
Update gatsby monorepo to v4.1.0 (#9367)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

benhaynes committed:
remove default color options from labels display (#9362)

azrikahar committed:
Translate system fields when creating new collections (#8104)

* translate system fields when create collections

* handle '$t:' translations only on interface

Co-authored-by: Jose Varela <joselcvarela@gmail.com>

renovate[bot] committed:
Update vue monorepo to v3.2.21 (#9341)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency mime to v2.6.0 (#9370)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency marked to v3.0.8 (#9359)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency openid-client to v5 (#9202)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

benhaynes committed:
add live chat to public docs (#9373)

rijkvanzanten committed:
Docs live chat tweak  (#9374)

* add live chat to public docs

* moving snippet to correct element

Co-authored-by: Ben Haynes <ben@rngr.org>

paescuj committed:
Fix condition to translate directus collections (#9353)

* Fix condition to translate directus collections

* Check whether collection belongs to 'directus_collections'

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

renovate[bot] committed:
Update dependency knex to v0.95.13 (#9378)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency @types/ldapjs to v2.2.2 (#9377)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

azrikahar committed:
Set user token as unique (#9381)

azrikahar committed:
Update users.md (#9382)

renovate[bot] committed:
Update dependency mime to v3 (#9380)

rijkvanzanten committed:
New translations en-US.yaml (Russian) (#9388)

rijkvanzanten committed:
New translations en-US.yaml (Bulgarian) (#9402)

rijkvanzanten committed:
New translations en-US.yaml (Bulgarian) (#9405)

rijkvanzanten committed:
New translations en-US.yaml (Bulgarian) (#9409)

rijkvanzanten committed:
New translations en-US.yaml (Bulgarian) (#9413)

rijkvanzanten committed:
New translations en-US.yaml (Bulgarian) (#9415)

paescuj committed:
Enhance comments in CI workflow (#9408)

Toilal committed:
Fix hover effect on o2m and m2a lists (#9412) (#9414)

aidenfoxx committed:
Added user rebinding on reconnect in LDAP (#9339)

* Added reconnect handling code for LDAP

* Update api/src/auth/drivers/ldap.ts

* Improved error handling

* Updated re-bind so we can await it when required

* Added bind check before operations to ensure client has user

* Cleaned up reconnect handling in LDAP

* Minor cleanup

* Minor cleanup

paescuj committed:
Link to all (open & closed) issues in bug report (#9394)

joselcvarela committed:
fix corresponding field name (#9393)

aidenfoxx committed:
Update config-options.md (#9218)

* Update config-options.md

* Update config-options.md

renovate[bot] committed:
Update dependency pinia to v2.0.1 (#9428)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

jamescammarano committed:
Fixed SET NULL on directus_files for MSSQL (#9425)

* fixed migration on mssql

* removed useless migration

oreilles committed:
Use primary key as default sort in map layout (#9403)

paescuj committed:
Move renovate config into .github (#9404)

benhaynes committed:
remove module and collection overrides from role

renovate[bot] committed:
Update fullcalendar monorepo to v5.10.1 (#9386)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
Update dependency tinymce to v5.10.1 (#9385)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

azrikahar committed:
Add placeholder to Language Indicator Field for first Translations setup (#9337)

* Hide Language Indicator Field initially

* fix lint error

* revert hiding languageFIeld & add placeholder

Toilal committed:
Fix related value for alias typed fields (#9210) (#9401)

rijkvanzanten committed:
Allow dynamic user variables to be used with filter rules (cont.) (#9376)

* Move permissions extraction to accountability

* Fix permissions retrieval for public user

* Fetch user / role context in permissions middleware

* Remove unnecessary parseFilter

* Rename schemaCache to systemCache

* Add permissions caching

* Add system cache invalidation on permission changes

* Improve caching perf by reducing scope

* Add note to docs

* Clarify compatibility with conditional fields/filters

* Fix lint warning

* Allow nested vars in system-filter-input

* Add custom getter function that resolves arrays

* Add is-dynamic-variable util

* Export new util

* Cleanup parse filter

* Fix build

* Move debounce up to use-items

* Remove unused prop

* 🧹

* Fix input pattern usage w/ vars

* Remove debounce from search-input, increase throttle

renovate[bot] committed:
Update dependency pinia to v2.0.2 (#9436)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

nickrum committed:
Rework hook registration (#8027)

* Rework hook registration

* Remove event and action fields from hook payloads

* Move "error" action to "request.error" filter

* Emit meta and context objects in filters and actions

* Run filters sequentially

* Update hook templates

* Fix CLI hook test

* Also emit `<collection>.items.crud` when emitting `items.crud`.

* Update hook docs

Co-authored-by: Oreilles <oreilles.github@nitoref.io>
Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

rijkvanzanten committed:
Fix duplicate hubspot injection (#9437)

nickrum committed:
Clean up API exports (#9418)

* Move API entrypoint to typescipt

* Remove explicit cli exort from API

The "./*" export should export all subpaths, so explicitly exporting the cli shouldn't be necessary.

* Remove dedicated start script from API

This script isn't even included in the final npm package and it's only used by the test Dockerfile.

rijkvanzanten committed:
Tweak project-settings page (#9440)

rijkvanzanten committed:
New Crowdin updates (#9442)

* New translations en-US.yaml (Serbian (Latin))

* New translations en-US.yaml (Portuguese, Brazilian)

* New translations en-US.yaml (Spanish, Chile)

* New translations en-US.yaml (Thai)

* New translations en-US.yaml (Breton)

* New translations en-US.yaml (Spanish, Latin America)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Swedish)

* New translations en-US.yaml (Estonian)

* New translations en-US.yaml (Chinese Simplified)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Finnish)

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Catalan)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Hungarian)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Lithuanian)

* New translations en-US.yaml (Dutch)

* New translations en-US.yaml (Slovenian)

rijkvanzanten committed:
Rename Collections Modules to Content Module (#9441)

* Rename collections->content module

* Replace collection name

* Replace some loose ends

nickrum committed:
Add exports fields to all packages (#9443)

rijkvanzanten committed:
Rename activity->notifications module (#9446)

rijkvanzanten committed:
v9.0.0-rc.101

rijkvanzanten committed:
Update changelog.md

adanielyan commented:
Is there anything I can do to help move this forward?

rijkvanzanten committed:
New translations en-US.yaml (Italian) (#9458)

rijkvanzanten committed:
New translations en-US.yaml (Italian) (#9472)

benhaynes commented:
I realize that alpha may be a common use-case, but it does change the saved value output. I'd like to clearly split this out as an interface option: `Enable Alpha Channel` or `Enable Opacity` (toggle), and it should be **off** by default.

* When enabled, the dropdown labels should switch to: `RGBA` and `HSLA` to clarify
* The opacity should just be a fourth INT input directly after the three color options
* Ideally, I'd to have a `%` appended to the opacity input, eg: `56%`. This might require extending the core component to support though... and is not required for this.
* I'd like to fix the system popup, which shows at the top-left of the viewport for me (disconnected from the interface, see below)
* Let's wait on a slider until we can do it right. I'd like to have color+opacity like most design apps, but STILL keep the inputs (see below)

![Screen Shot 2021-11-04 at 11 40 23 AM](https://user-images.githubusercontent.com/522079/140365609-821bbecb-a44a-4d37-aa9c-be1dac48772d.png)

![Screen Shot 2021-11-04 at 11 39 43 AM](https://user-images.githubusercontent.com/522079/140367229-5734ce43-cc18-402f-9b8e-77d4ff87073f.png)

I'm happy to do a design for how this should look with _all_ of these features implemented, but that really depends on how far you'd like to take this. If this PR is just about adding opacity, then some of the above is all we need.

rijkvanzanten committed:
New translations en-US.yaml (Italian) (#9476)

rijkvanzanten committed:
Resolve calendar link to detail page (#9477)

rijkvanzanten committed:
Remove leading slash

paescuj committed:
Group Renovate updates & schedule weekly (#9479)

Toilal committed:
Fix invalid collection for Interface Display Template for M2M relationships (#9420)

* Fix invalid collection for Interface Display Template (#9419)

* use o2m ollection on m2m interface

* apply same fix on files

Co-authored-by: Jose Varela <joselcvarela@gmail.com>

joselcvarela committed:
SDK: refactor constructors (#9080)

* refactor: more intuitive interfaces

* refactor: simpler refresh
before: on every request we were debouncing a refresh request
after: call refresh only once before now() + 'expires'

* refactor: prefix on base storage

* fixup! refactor: simpler refresh before: on every request we were debouncing a refresh request after: call refresh only once before now() + 'expires'

* refactor: simpler axios transport
before: handle auth headers
after: auth headers are handled on directus instance

* refactor: simpler usage of Directus constructor

* fixup! refactor: simpler refresh before: on every request we were debouncing a refresh request after: call refresh only once before now() + 'expires'

* refactor: fix tests based on previous changes

* refactor: better auth constructor
before: depends on SDK instance
after: depends on Transport and Storage instance

* accept staticToken from auth

* make transport and storage as optional on options

* fix type auth refresh

* simplify transport

* fix test for previous changes

* improve auth class

* revert some IAuth props because tests

* allow to force memory of localstorage on storage

* add tests for previous change

* document everything and simplify some things

* fix override headers on request

* better name typing

* fix private axios

* removed boolean from CLI auth.refresh()

* fix missing url in some examples

* soem grammar updates

Co-authored-by: Jay Cammarano <jay.cammarano@gmail.com>
Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

paescuj committed:
Fixing some type errors in app (#9466)

* Fixing some type errors in app

* Remove unused import

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

Toilal committed:
Fix render template for number template parts (#9406) (#9407)

azrikahar committed:
fix docs homepage header responsiveness (#9450)

rijkvanzanten committed:
New translations en-US.yaml (Italian) (#9482)

rijkvanzanten committed:
Fix nested system-interface-options usage (#9483)

* Improve null check in list

* Fix options syncing in system-interface-options

rijkvanzanten committed:
Fix m2a relations on editing field (#9484)

Fixes #9463

rijkvanzanten committed:
Fix presentation-links interfaces (#9485)

Fixes #9457

joselcvarela committed:
fix docs about sdk refactor (#9486)

nickrum committed:
Clarify hook register function parameter descriptions in docs (#9489)

* Clarify hook register function parameter descriptions in docs

* Update package-lock.json

* Update docs/extensions/hooks.md

Co-authored-by: Pascal Jufer <pascal-jufer@bluewin.ch>

Co-authored-by: Pascal Jufer <pascal-jufer@bluewin.ch>

paescuj committed:
Schedule Renovate on a daily basis for now (#9488)

faridsaud committed:
Use hash instead of random for default index name (#9448)

* [Fix]: Use hash instead of random for default index name

* Move hash function to separate util file

* Reduce max size of index name to 60 from 64

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

rijkvanzanten committed:
Remove collection listing option from role settings (#9490)

* Fixes #8920

* Revert changes in seeds / migrations

* Add new migration to drop column

Co-authored-by: Yannick Mol <yannick@molmediasolutions.nl>

rijkvanzanten committed:
Remove beta flag from insights & enable by default (#9491)

rijkvanzanten committed:
New Crowdin updates (#9493)

* Update source file en-US.yaml

* New translations en-US.yaml (Serbian (Latin))

* New translations en-US.yaml (Portuguese, Brazilian)

* New translations en-US.yaml (Indonesian)

* New translations en-US.yaml (Persian)

* New translations en-US.yaml (Spanish, Chile)

* New translations en-US.yaml (Thai)

* New translations en-US.yaml (Breton)

* New translations en-US.yaml (Chinese Traditional)

* New translations en-US.yaml (Spanish, Latin America)

* New translations en-US.yaml (Russian)

* New translations en-US.yaml (Polish)

* New translations en-US.yaml (Portuguese)

* New translations en-US.yaml (Swedish)

* New translations en-US.yaml (Turkish)

* New translations en-US.yaml (Estonian)

* New translations en-US.yaml (Vietnamese)

* New translations en-US.yaml (Chinese Simplified)

* New translations en-US.yaml (French)

* New translations en-US.yaml (Finnish)

* New translations en-US.yaml (Spanish)

* New translations en-US.yaml (Arabic)

* New translations en-US.yaml (Bulgarian)

* New translations en-US.yaml (Catalan)

* New translations en-US.yaml (Czech)

* New translations en-US.yaml (Danish)

* New translations en-US.yaml (German)

* New translations en-US.yaml (Hungarian)

* New translations en-US.yaml (Italian)

* New translations en-US.yaml (Japanese)

* New translations en-US.yaml (Korean)

* New translations en-US.yaml (Lithuanian)

* New translations en-US.yaml (Dutch)

* New translations en-US.yaml (Slovenian)

nickrum committed:
Clean up interface options type (#9447)

rijkvanzanten committed:
Move union query application to applyQuery, fix where clause (#9494)

* Move union query application to applyQuery, fix where clause

Fixes #9228

* Handle case where union IDs length = 0

* Return modified db query

* Apply union last

rijkvanzanten committed:
Don't show all migrations logged on init (#9496)

tylerforesthauser committed:
Small fixes for Hooks documentation (#9497)

Remove erroneous closing bracket and remove index position from input as it should be an object in this context, not an array.

adanielyan commented:
I think the first 2 points are doable. As for the system picker, unfortunately none of the current browsers support alpha channel in the system picker AFAIK. Alternatively we could use a third party picker instead of the system one, but I think it me be too much for such a simple feature.

I'll try to implement the first two recommendations though.

benhaynes commented:
Awesome! For the system picker, we already have that (if you click on the color in the input), but it shows up in the wrong location within the App. Was just hoping we could get that pinned to the interface.

But that brings up a good point. If the system interface doesn't support opacity, then using that input method wouldn't work if you had "Enable Alpha Channel" on. I wonder how these two inputs will play together. 🤔

adanielyan commented:
In my implementation that is in this PR, the color picker is only responsible for color, while the opacity is controlled by a separate slider. So the two work nicely together. See the video here: https://github.com/directus/directus/pull/9224#issuecomment-954941983

rijkvanzanten committed:
Prevent negative hashes from being generated (#9501)

Fixes #9499

oreilles committed:
Properly handle M2A fields in fieldStore and useFieldTree (#9432)

* Properly handle M2A fields in fieldStore and useFieldTree

* Fix addNode

* Rewrote use-field-tree

* Remember visited paths

* Fix error whith undefined relation.meta

nickrum committed:
Fix importing the SDK from a Node ESM environment (#9502)

oreilles committed:
Update geometric types and patch new field flow. (#9397)

* Update geometric types and patch new field flow.

* Add migration

* Fixed migrations

* Also fixed migrations

* Update migration ID

* Cleanup type selector a bit

* Add missing fallback interface/display for new types

Co-authored-by: rijkvanzanten <rijkvanzanten@me.com>

rijkvanzanten committed:
v9.0.0

rijkvanzanten committed:
Update changelog.md

paescuj committed:
Revert "Schedule Renovate on a daily basis for now (#9488)" (#9508)

This reverts commit 42512b88740f66a25ec861a85ae8bf494410bd33.

paescuj committed:
Update Docker installation documentation (#9514)

* Reflect new stable version and available Docker tags
* Clean-up

paescuj committed:
Add note about Directus version in bug report (#9517)

paescuj committed:
Remove quotes on string values for raw display (#9522)

azrikahar committed:
fix repeater field names title format (#9504)

paescuj committed:
Remove note about RC in the readme (#9513)

paescuj committed:
Update package-lock.json & fix warning from eslint (#9509)

* Update package-lock.json

To reflect latest release

* Fix warning from eslint

nickrum committed:
Include the ESM entrypoint when publishing to npm (#9535)

benhaynes committed:
divider title placeholder (#9532)

dimitrov-adrian committed:
Fix missing and wrong translations (#9537)

benhaynes committed:
add installation to docs homepage

benhaynes committed:
add translated field docs

benhaynes committed:
formatting

benhaynes committed:
Data model dense (#9558)

* remove extra popup padding

* tweak sidebar spacing

* use dense on data model collection listing

benhaynes committed:
small text update (#9554)

azrikahar committed:
Minor docs fixes (#9553)

joselcvarela committed:
Fix lose data on M2M (#9548)

* save initial items of m2m relation

* merge initial, draft and selected on new selection

renovate[bot] committed:
Update NPM dependencies (non-major) (#9506)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

azrikahar committed:
fix elements z-index higher than app header bar (#9530)

benhaynes committed:
selector for override (#9562)

paescuj committed:
Fix pr title for major updates from renovate (#9594)

benhaynes committed:
update in-app docs nav (#9610)

* update in-app docs nav

* remove section from in-app docs

benhaynes committed:
fix in-app docs homepage (#9611)

benhaynes committed:
fixes: #9568 (#9612)

Commit:
added switch

Commit:
merge conflict

adanielyan commented:
I added switch as suggested above.

https://user-images.githubusercontent.com/861777/140844873-79db3c35-e8c1-47a2-87c6-7ceb753d8816.mov

Commit:
small bug fixes and stacked layout support

adanielyan commented:
And the stacked layout

https://user-images.githubusercontent.com/861777/140847867-2ba15abe-701b-4e94-bb9b-1585b1b6d0f5.mov

benhaynes commented:
Hey @adanielyan — **so** sorry for not being clearer, I meant we should put the "Include Alpha" as an _Interface Option_... not an option in the interface. 😬 

The interface options are set by administrators only (and are written in the interface's `indx.ts` file), so they can control the saved format... which an end user should not be controlling.

Other than that, this is looking really nice!! ❤️

Commit:
added Alpha config to color inteface

adanielyan commented:
> Hey @adanielyan — **so** sorry for not being clearer, I meant we should put the "Include Alpha" as an _Interface Option_... not an option in the interface. 😬
> 
> The interface options are set by administrators only (and are written in the interface's `indx.ts` file), so they can control the saved format... which an end user should not be controlling.
> 
> Other than that, this is looking really nice!! ❤️

OK, that's an easy fix.

https://user-images.githubusercontent.com/861777/140975512-0e3a77c5-6f56-466e-b891-79bcc75eb03b.mov

If that's not what you want I give up :)

Commit:
set default val of showAlpha

benhaynes commented:
Haha, getting closer! But if the Alpha Interface Option (in Settings) is enabled, we don't want/need a second toggle on the interface itself. So let's remove the toggle in the dropdown and make the alpha slider full-width. 👍

Commit:
removed alpha switch

adanielyan commented:
I hope this is the last one :)

https://user-images.githubusercontent.com/861777/141142676-ba4a7908-a62f-4902-920b-1af1ce0516ab.mov

benhaynes commented:
Perfect, thank you!! The only question now is what should we do with that slider when there is no value (`NULL`). Maybe we conditionally don't show the slider, or use some sort of "default" background color? Thoughts @rijkvanzanten ?

adanielyan commented:
I thought about that too... Even hid the slider when the value is null, but then the interface looked inconsistent with the slider suddenly appearing after selecting a color, so I reverted it back.

benhaynes commented:
Yeah, makes sense. The slider is "extra", as we have the input above, so we could simply not include it in the interface. Otherwise, I think we'd need some sort of indeterminate state for the slider. Instead of you going back and forth, I can provide a design (not sure when)... or we can remove the slider for now.

Commit:
added slider bg

benhaynes commented:
Yeah, I like that! Better for transparency too. Is that slider background using the global border radius variable? It looks a little squared off... maybe we just make it completely rounded on the ends?

adanielyan commented:
And in the light mode

https://user-images.githubusercontent.com/861777/141150511-e2fe68da-35bc-41b1-b2f5-c5c8a7e19a7a.mov

adanielyan commented:
> Yeah, I like that! Better for transparency too. Is that slider background using the global border radius variable? It looks a little squared off... maybe we just make it completely rounded on the ends?

Not sure... I didn't change the border radius of the standard slider.

adanielyan commented:
It seems it's the border radius of input element, 2px.

Commit:
slider css fix for safari

Commit:
slider rounded edges

Commit:
slider rounded edges

Commit:
slider rounded edges

Commit:
slider rounded edges

adanielyan commented:
The slider component is broken in Safari. The reason is that the `appearance: none;` is overridden by Safari's `-webkit-appearance: slider-horizontal;`. Adding `-webkit-appearance: none;` in slider's css fixes the problem. However on every commit it is converted to `appearance: none;` which doesn't do the trick. Any ideas how to fix that?

benhaynes commented:
Not sure about the slider in Safari... I'll leave that to the engineers. But the last (small) request, is to decrease the space between the slider and the color chip options. Looking great, thank you for all this awesome work!

rijkvanzanten commented:
I think the small border radius is due to the fact that the slider track is made to be 4px high, at which size the border radius is 50%, however, with this thicker track that border radius no longer fits 🙂 

> The reason is that the appearance: none; is overridden by Safari's -webkit-appearance: slider-horizontal;. Adding -webkit-appearance: none; in slider's css fixes the problem. However on every commit it is converted to appearance: none; which doesn't do the trick. Any ideas how to fix that?

Good one.. I believe that's stylelint that's slightly too aggressive. Here's to hoping the Safari team hurries up with the unprefixed support; it's already support in Technology Preview after all! 😄

adanielyan commented:
Here you go, plus rounded edges

![image](https://user-images.githubusercontent.com/861777/141388333-8a4251b1-cd4e-4a83-b0b6-cfb55f518694.png)

![image](https://user-images.githubusercontent.com/861777/141388370-6ddc2f5d-845a-4de8-a473-32daabc7a082.png)

Commit:
padding for slider

adanielyan commented:
> I think the small border radius is due to the fact that the slider track is made to be 4px high, at which size the border radius is 50%, however, with this thicker track that border radius no longer fits 🙂

In fact the rounded corners were added by the browser (Edge). But I made them fully round anyway. I had to add that to v-slider component, but it will not affect the regular slider if `--input-bg-image` css var is not explicitly provided in the slider props.

> Good one.. I believe that's stylelint that's slightly too aggressive. Here's to hoping the Safari team hurries up with the unprefixed support; it's already support in Technology Preview after all! 😄

If it's good for you it's good for me. I don't use Safari anyway! 😄

Commit:
Merge branch 'main' of github.com:directus/directus into color-with-alpha

adanielyan commented:
Hey @benhaynes, is there anything else blocking this PR? If not, maybe it can be merged? Thanks!

benhaynes commented:
Just the time it takes to have engineering formally review and merge. We've been abnormally busy on cloud the past few weeks... but should have more time starting next week. 👍

adanielyan commented:
Sure, understood. Just wanted to make sure there's nothing else to add/fix.

Commit:
renamed Allow Alpha to Opacity

rijkvanzanten commented on app/src/interfaces/select-color/select-color.vue:
```diff
@@ -30,14 +36,38 @@
 				<v-select v-model="colorType" :items="colorTypes" />
 			</div>
-			<template v-if="colorType === 'RGB'">
+			<template v-if="(colorType === 'RGB') | (colorType === 'RGBA')">
```
```suggestion
			<template v-if="colorType === 'RGB' || colorType === 'RGBA'">
```

rijkvanzanten commented on app/src/interfaces/select-color/select-color.vue:
```diff
@@ -48,10 +78,22 @@
+				/>
 			</template>
-			<template v-if="colorType === 'HSL'">
+			<template v-if="(colorType === 'HSL') | (colorType === 'HSLA')">
```
```suggestion
			<template v-if="colorType === 'HSL' || colorType === 'HSLA'">
```

rijkvanzanten commented on app/src/interfaces/select-color/select-color.vue:
```diff
@@ -63,8 +105,41 @@
+						'--background-page': 'none',
+						'--v-slider-fill-color': 'none',
+						'--v-slider-thumb-color': 'var(--foreground-normal)',
+						'--input-bg-image':
+							'linear-gradient(to right, transparent,' +
+							(hex && hex.length === 9 ? hex.slice(0, -2) : hex ? hex : 'transparent') +
+							'), url(\'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAMUlEQVQ4T2NkYGAQYcAP3uCTZhw1gGGYhAGBZIA/nYDCgBDAm9BGDWAAJyRCgLaBCAAgXwixzAS0pgAAAABJRU5ErkJggg==\')',
```
Great solution here! One little nitpick, lets rename `--input-bg-image` to `--v-slider-track-background-image`.

rijkvanzanten commented on app/src/interfaces/select-color/select-color.vue:
```diff
@@ -184,30 +282,50 @@ export default defineComponent({
+					let alpha = Math.round(255 * color.value.alpha())
+						.toString(16)
+						.toUpperCase();
+					alpha = alpha.length === 1 ? `0${alpha}` : alpha;
```
Lets use `padStart` here

rijkvanzanten reviewed (changes requested):
Couple of tiny nitpicks / tweaks. Otherwise great addition. Thanks @adanielyan! 😄

adanielyan committed:
Update app/src/interfaces/select-color/select-color.vue

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

adanielyan committed:
Update app/src/interfaces/select-color/select-color.vue

Co-authored-by: Rijk van Zanten <rijkvanzanten@me.com>

Commit:
implemented suggestions made by @rijkvanzanten

adanielyan commented:
Thanks for suggestions! They have been implemented.

adanielyan commented:
Hi, is there any other action I need to take before this can be merged? Sorry, I am not very familiar with the process.

benhaynes commented:
No, you're good. We've just been super busy with other releases and the holidays. We'll try to get this reviewed/merged ASAP.

adanielyan commented:
Great, thanks @benhaynes!

rijkvanzanten committed:
Merge branch 'main' into color-with-alpha
